doc/sphinxext/move_gallery_links.py

```diff
@@ -0,0 +1,183 @@
+"""
+This script intends to better integrate sphinx-gallery into pydata-sphinx-theme. In
+particular, it moves the download links and badge links in the footer of each generated
+example page into the secondary sidebar, then removes the footer and the top note
+pointing to the footer.
+
+The download links are for Python source code and Jupyter notebook respectively, and
+the badge links are for JupyterLite and Binder.
+"""
```

Comment:
```suggestion

Currently this is achieved via post-processing the HTML generated by sphinx-gallery.
A possible fix upstream to avoid the need for post-processing is discussed at:
https://github.com/sphinx-gallery/sphinx-gallery/issues/1258
"""
```